apps/gallery/js/thumbnail_list.js

```diff
@@ -35,9 +35,9 @@ function ThumbnailList(groupClass, container) {
   this.groupClass = groupClass;
   this.container = container;
 
-  // If the locale changes while we're running, then we need to
-  // re-localize the dates in the group headers
-  navigator.mozL10n.ready(this.localize.bind(this));
+  // Localize dates in group headers and thumbnail item
+  // description when l10n resources are loaded
+  navigator.mozL10n.once(this.localize.bind(this));
```

Comment:
If you do it like this, then we may get localized twice on startup once here, and once for the ready call in startup.js.

thumbnail_item and thumbnail_date_group already have code to localize themselves if readyState === complete when they are created. So we only need to handle the case where the thumbnail list is being created before the locale data is loaded.  Assuming you make the ready() call right after calling new ThumbnailList, I think that is all we need.